core/src/main/java/org/opensearch/sql/utils/IPUtils.java

```diff
@@ -0,0 +1,33 @@
+package org.opensearch.sql.utils;
+
+import com.google.common.net.InetAddresses;
+import org.opensearch.sql.data.model.ExprBooleanValue;
+import org.opensearch.sql.data.model.ExprValue;
+
+import java.util.Arrays;
+
+public class IPUtils {
+
+    /**
+     * Returns whether the given IP address is within the specified IP address range.
+     * Supports both IPv4 and IPv6 addresses.
+     *
+     * @param addressExprValue IP address (e.g. "198.51.100.14" or "2001:0db8::ff00:42:8329").
+     * @param rangeExprValue   IP address range in CIDR notation (e.g. "198.51.100.0/24" or "2001:0db8::/32")
+     * @return true if address is in range; else false
+     */
+    public static ExprBooleanValue isAddressInRange(ExprValue addressExprValue, ExprValue rangeExprValue) {
+
+        try {
+            byte[] addressBytes = InetAddresses.forString(addressExprValue.stringValue()).getAddress();
+
+            String[] rangeFields = rangeExprValue.stringValue().split("/");
+            int prefixLengthBytes = Integer.parseInt(rangeFields[1]) / Byte.SIZE;
+            byte[] rangeBytes = Arrays.copyOfRange(InetAddresses.forString(rangeFields[0]).getAddress(), 0, prefixLengthBytes);
+
+            return ExprBooleanValue.of(Arrays.equals(addressBytes, 0, prefixLengthBytes, rangeBytes, 0, prefixLengthBytes));
+        } catch (Exception e) {
```

Comment:
I think we should explicitly check that the string is one of the permitted formats rather than throw/catch potential exceptions like index out of bounds exceptions.

Reply:
I believe that this comment has been overtaken by changes.